src/components/Product/Product.js

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from 'react';
+import firebase from '@firebase/app';
+function Product() {
+  const [option, setOption] = useState('');
+  const [data, setData] = useState({
+    name: '',
+    lastDate: '',
+  });
+  const addProduct = (event) => {
+    event.preventDefault();
+    console.log('send data...' + data.name, data.lastDate, option);
+    firebase
+      .firestore()
+      .collection('products')
+      .add({
+        name: data.name,
+        lastDate: data.lastDate,
+        date: option,
+        token: token(),
+      })
+      .then((quantityAfter) => {
+        console.log('the amount was saved successfully', quantityAfter);
+      })
+      .catch((error) => {
+        console.log('error', error);
+      });
+  };
+  const handleInputChange = (event) => {
+    setData({
+      ...data,
+      [event.target.name]: event.target.value,
+    });
+  };
+  var rand = function () {
+    return Math.random().toString(36).substr(2);
+  };
+  var token = function () {
+    return rand() + rand();
+  };
+  return (
+    <div>
+      <form>
+        <div>
+          <label>
+            Name:
+            <input type="text" onChange={handleInputChange} name="name" />
```

Comment:
good to see destructuring assignment but I would like to see the update directly because it reduce use this function handleInputChange onChange={(e) => setName(e.target.name)}